### PR TITLE
Pado dev branch

### PR DESCRIPTION
**Pull request by dpastierovic**
This branch contains fixes to bugs:
https://github.com/ERNICommunity/ERNI-Personal-Budget-Application/issues/42
https://github.com/ERNICommunity/ERNI-Personal-Budget-Application/issues/40
and implementation for enhancements:
https://github.com/ERNICommunity/ERNI-Personal-Budget-Application/issues/32
https://github.com/ERNICommunity/ERNI-Personal-Budget-Application/issues/41

### CODE REVIEW

File: client/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (menyet) [inline client/.vscode/launch.json]: Do we need this file?

File: client/src/app/services/user.service.ts [hunk 2]
```diff
@@ -13,6 +13,10 @@ export class UserService {
     constructor(private http: HttpClient, private serviceHelper: ServiceHelper, private configService: ConfigService) {
     }
 
+    public registerUser(user: any): Observable<any> {
```
Comment (menyet) [inline client/src/app/services/user.service.ts]: the argument user is not used

File: client/src/app/login-callback/oauth-callback.component.ts [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
     ngOnInit() {
         if (!this.adalService.userInfo) {
             this.router.navigate(['login']);
-        } else {
-            this.router.navigate(['my-budget']);
+        } else { 
+            let sub = this.adalService.userInfo.profile['sub'];
+            let firstName = this.adalService.userInfo.profile['given_name'];
+            let lastName = this.adalService.userInfo.profile['family_name'];
+            let upn = this.adalService.userInfo.profile['upn'];
```
Comment (menyet) [inline client/src/app/login-callback/oauth-callback.component.ts]: the registerUser method does not need any arguments, you can remove these lines

File: server/ERNI.PBA.Server.Host/Startup.cs [hunk 4]
```diff
@@ -78,35 +80,28 @@ public void ConfigureServices(IServiceCollection services)
...
-                                };
-                                db.Users.Add(user);
-                                db.SaveChanges();
-                            }
-
-                            var claims = new List<System.Security.Claims.Claim>();
-                            claims.Add(new System.Security.Claims.Claim(Claims.Id, user.Id.ToString()));
-                            claims.Add(new System.Security.Claims.Claim(Claims.FirstName, user.FirstName));
-                            claims.Add(new System.Security.Claims.Claim(Claims.LastName, user.LastName));
+                                new Claim(Claims.FirstName, context.Principal.Claims.Single(c => c.Type == "given_name").Value),
```
Comment (menyet) [inline server/ERNI.PBA.Server.Host/Startup.cs]: these claims we already have, we do not need to create them again

Comment (dpastierovic) [reply server/ERNI.PBA.Server.Host/Startup.cs]: We do have them, but only in the token. Each time we want to use them, we need to write LINQ query. This stores them in statically typed class for easier use in the rest of the application.

**Pull request merged**